### PR TITLE
Completeness for 0.1

### PR DESCRIPTION
**Pull request by judepereira**

### CODE REVIEW

**Comment by stevepeak** [comment]
Wrote some high-level docs on Events. Need some feedback on organization of concepts before digging lower.

----

# Events

... description ...

[[toc]]

### Rules
1. A service **MAY** have commands that are event-driven.
1. A command **MAY** have many events.
1. Event **MAY** have arguments which are provided during subscription.
1. An event **MUST** include subscribe and unsubscribe protocols.
1. Event titles **MUST** match the pattern `[\w\-]{3,45}`

### Platform Workflow
1. Platform **subscribes** to an event.
1. Service **publishes** an event.
1. Platform **handles** the event.
1. Repeat #2 and #3 until Platform **unsubscribes**.

#### Examples
The services below have event-based commands.

- [Slack Bot](https://github.com/microservice/slack/blob/461617252e7296a075d89a5f936c9691a93f62e8/microservice.yml#L6)
- [Steaming Twitter tweets]()

### Configuration

```yaml
commands:
  stream:
    run: # Command to start an http/rpc service to receive subscriptions
      ...
    events:
      tweet:
        arguments: # Map of arguments provided during subscription
          ...
        output:    # Describing what data this event publishes
          ...
```

... Table showing structure ...


#### `run`
A command **MAY** provide a `run` block that describes how to start a server to subscribe to.
Otherwise the [lifecycle]() server is used.

#### `events`
Events are subscribed to and published by the service back to the Platform.

File: overview/README.md [hunk 1]
```diff
@@ -16,3 +21,6 @@ A `microservice.yml` file **MUST** be provided which outlines the command struct
 - **MAY** - Optional for a service to contain this functionality
 
 See [RFC 2119](https://tools.ietf.org/html/rfc2119) for details
+
+## Next
```
Comment (stevepeak) [inline overview/README.md]: I suggest making these H3 `###` to remove them from the side bar.

File: schema/actions/README.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
+# Actions
+Defining actions is a way to organize which APIs are exposed 
+and how they are used within an application or platform.
+
+Properly defining actions can assist with the following:
+  - Service discovery
+  - Documentation
+  - Operational insight
+
+Services **MUST** define actions and arguments that can be used.
```
Comment (stevepeak) [inline schema/actions/README.md]: Services **MUST** list actions that can be executed.

I feel like arguments does not fit into this sentence well.

File: schema/actions/README.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+- Output 
+
+
+**Sample**
+
+```yaml {4-5}
+actions:
+  convert:
+    help: Convert a currency into another currency
+    http/format:
```
Comment (stevepeak) [inline schema/actions/README.md]: This is very confusing at first. Was not clear that it was either-or.

```yaml
http:  # choose: http, format, rpc
   ...
```

File: schema/actions/README.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+Within a named action, the following fields are available:
+
+<json-table>
+<p>
+{
+    "help": {
+        "desc": "A human friendly description for this action"    
+    }, 
+    "http": {
+        "desc": "This is described in depth [here](/schema/interface/#http)"
```
Comment (stevepeak) [inline schema/actions/README.md]: `This is here` sound odd.

I think a simple `Read more` does the job.

**Comment by judepereira** [comment]
- [ ] startup is optional - defaults to tail -f /dev/null
- [ ] commands can be a string or in an array format - array is preferred
- [ ] grep and deprecate OMG_ENDPOINT
- [ ] remove "server to platform"

**Pull request merged**